starter-core/src/main/java/io/micronaut/starter/feature/microstream/MicroStreamRest.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2017-2021 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.starter.feature.microstream;
+
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.starter.application.ApplicationType;
+import io.micronaut.starter.application.generator.GeneratorContext;
+import io.micronaut.starter.build.dependencies.Dependency;
+import io.micronaut.starter.feature.Category;
+import io.micronaut.starter.feature.Feature;
+import io.micronaut.starter.feature.FeatureContext;
+import jakarta.inject.Singleton;
+
+@Singleton
+public class MicroStreamRest implements Feature {
+
+    public static final String NAME = "microstream-rest";
+    private static final String MICRONAUT_MICROSTREAM_GROUP_ID = "io.micronaut.microstream";
+
+    private final MicroStream microStream;
+
+    MicroStreamRest(MicroStream microStream) {
+        this.microStream = microStream;
+    }
+
+    @Override
+    @NonNull
+    public String getName() {
+        return NAME;
+    }
+
+    @Override
+    public boolean isPreview() {
+        return true;
+    }
+
+    @Override
+    public String getTitle() {
+        return "MicroStream REST";
+    }
+
+    @Override
+    @NonNull
+    public String getDescription() {
+        return "Adds the MicroStream REST API to your Micronaut project for development";
+    }
+
+    @Override
+    public boolean supports(ApplicationType applicationType) {
+        return true;
+    }
+
+    @Override
+    public String getCategory() {
+        return Category.DATABASE;
+    }
+
+    @Override
+    public void apply(GeneratorContext generatorContext) {
+        generatorContext.getConfiguration().put("microstream.rest.enabled", "true");
```

Comment:
don't enable it by default. User should have to opt-in. I don't want people to inadvertently expose their contents of their storage.